ObjcVoiceQuickstart/ViewController.m

```diff
@@ -391,7 +403,7 @@ - (void)call:(TVOCall *)call didFailToConnectWithError:(NSError *)error {
     NSLog(@"Call failed to connect: %@", error);
     
     self.callKitCompletionCallback(NO);
-    [self performEndCallActionWithUUID:call.uuid];
+    [self.callKitProvider reportCallWithUUID:call.uuid endedAtDate:[NSDate date] reason:CXCallEndedReasonFailed];
```

Comment:
It was nice earlier that we were ending the call same way, now we have two different methods. Can we use 
use `reportCallWithUUID:call endedAtDate: reason` in `performEndCallActionWithUUID`?

Reply:
`performEndCallActionWithUUID` is the method used for initiating the disconnect from the app UI, while the `call:didDisconnectWithError:` callback method is invoked when the call is eventually disconnected by the action. That's why we use the `userInitiatedDisconnect` variable to determine whether we should inform CallKit how the call ended. Similarly, the `call:didFailToConnectWithError:` is where we report the terminating state of the call.
There is a bug in the original implementation here, where we will try to tell CallKit to end a call while this call, based on the API contract of this callback, was never connected, therefore CallKit will tell us that the application is ending a call with an invalid UUID.